sparkle/platform/output.py

```diff
@@ -0,0 +1,298 @@
+        """
+        self.configured_PAR10 = configured_par10
+        self.default_PAR10 = default_par10
+
+
+class ValResults:
+    """Class that stores validation information and results."""
+    def __init__(self: ValResults, solver: str, configuration: str, instance_set: str,
+                 results: list[list[str, Status, float, float]]) -> None:
+        """Initalize ValResults.
+
+        Args:
+            solver: The name of the solver
+            configuration: The configuration being used
+            instance_set: The set of instances
+            results: Validation results in the format:
+                [["instance", "status", "quality", "runtime"]]
+        """
+        self.solver = solver
+        self.configuration = configuration
+        self.instance_set = instance_set
+        self.result_header = ["instance", "status", "quality", "runtime"]
+        self.result_vals = results
+
+
+class ConfigurationResults:
+    """Class that aggregates configuration results."""
+    def __init__(self: ConfigurationResults, performance: ConfigurationPerformance,
+                 res_def: ValResults, res_conf: ValResults) -> None:
+        """Initalize ConfigurationResults.
+
+        Args:
+            performance: The performance of the default and configured solver
+            res_def: The default results
+            res_conf: The configured results
+        """
+        self.performance = performance
+        self.configured_results = res_conf
+        self.default_results = res_def
+
+
+class ConfigurationOutput:
+    """Class that collects configuration data and outputs it a json format."""
+    solver: Solver = None
+    configurator: Configurator = None
+    best_config: str = None
+    training: ConfigurationResults = None
+    test: ConfigurationResults = None
+
+    instance_set_test: InstanceSet = None
+
+    def __init__(self: ConfigurationOutput, path: Path, solver_name: str,
+                 configurator: Configurator, penalty_multiplier: int,
+                 output: Path = None) -> None:
+        """Initialize Configurator Output class.
+
+        Args:
+            path: Path to configuration directory
+            solver_name: Name of the solver that was used
+            configurator: The configurator that was used
+            penalty_multiplier: penalty multiplier that is applied to the par performance
+            output: Path to the output directory
+        """
+        self.solver = resolve_object_name(solver_name,
+                                          gv.solver_nickname_mapping,
+                                          gv.solver_dir, Solver)
+        self.configurator = configurator
+        self.penalty_multiplier = penalty_multiplier
+        self.directory = path
+
+        if output is None:
+            output = path / "Analysis" / "configuration.json"
+        self.output = output
+
+        # TODO: Fix this spaghetti code to find the scenario file
```

Comment:
Agreed, perhaps you can just pass the ConfigurationScenario object directly

Reply:
Question is where I get it from
It is implemented this way not because it was the easiest way, but because you told me to implement in a way where I read the most current version of scenario and not one that can be accessed over settings or some other way that might not contain changes by the user

Reply:
Is this resolved now?

Reply:
No as I don't know where you want me to read the scenario object from

Reply:
I talken to Jeroen about it and he thinks that this isn't really something that fits into the scope of this ticket, which I agree to
If getting the scenario settings is such a problem/ reading the global settings is to unreliable, then that should be a ticket itself. I added a comment about that to Jira